cluster-install/install.sh

```diff
@@ -256,13 +259,40 @@ get_connector_namespace() {
     echo "${NAMESPACE:-"${__DEFAULT_CONNECTOR_NAMESPACE}"}"
 }
 
+existing_trident_can_be_modified() {
+    [ "$DO_NOT_MODIFY_EXISTING_TRIDENT" == "true" ] && return 1
+    return 0
+}
+
+existing_trident_needs_modifications() {
+    if [ "$_TRIDENT_COLLECTION_STEP_CALLED" != "true" ]; then
+        fatal "this function should not be called until existing Trident information has been collected"
+    fi
+    trident_is_missing && return 1
+
+    components_include_trident && trident_image_needs_upgraded && return 0
+    components_include_trident && trident_operator_image_needs_upgraded && return 0
+    components_include_acp && acp_image_needs_upgraded && return 0
+    components_include_acp && ! acp_enabled && return 0
+
+    return 1
+}
+
 trident_is_missing() {
+    if [ "$_TRIDENT_COLLECTION_STEP_CALLED" != "true" ]; then
+        fatal "this function should not be called until existing Trident information has been collected"
+    fi
     [ -z "$_EXISTING_TRIDENT_NAMESPACE" ] && return 0
     return 1
 }
 
-should_skip_trident_upgrade() {
-    [ "$SKIP_TRIDENT_UPGRADE" == "true" ] && return 0
+
+trident_will_be_installed_or_modified() {
```

Comment:
With the number of variables involved in deciding whether to do anything with Trident grew, it became harder and harder to understand how it all worked and when to do what. This function is the end-all-be-all "are we doing anything with Trident or not, regardless of what it is" and gates various Trident-related functionality.